0011-interactive-transport-protocol/0011-interactive-transport-protocol.md

```diff
@@ -0,0 +1,104 @@
+# Interactive Transport Protocol (ITP)
+
+## Preface
+
+This document specifies the Interactive Transport Protocol. It is a host-to-host protocol for Interledger payment requests, which include receiver-generated conditions. This condition protocol enables receivers to verify that an incoming transfer matches a payment request previously created by them, without the receiver needing to keep a record of all outstanding payment requests.
+
+## Introduction
+
+### Motivation
+
+The "Universal" mode of the Interledger Protocol uses holds based on Crypto Conditions and expiries to ensure a sender's funds are delivered to the destination or returned to the sender's account. (For more on the Interledger Protocol see [IL-RFC-3](../0003-interledger-protocol).)
+
+A common approach is for conditions to be generated and fulfilled by the receivers of Interledger transfers. This document specifies one such approach that enables receivers to verify that an incoming transfer matches a payment request previously created by the receiver, without the receiver needing to keep a record of all outstanding payment requests.
+
+### Scope
+
+This document defines one approach for receiver-generated conditions. Since the condition is opaque to all parties aside from the receiver, the receiver may use any condition they choose. However, this specification includes best practices that receivers SHOULD take into consideration.
+
+Communicating payment requests from the receiver to the sender is *not* included in this protocol and is to be handled by higher level protocols.
+
+### Interfaces
+
+This protocol is used by higher-level protocols for the receiver to generate conditions and payment requests. The sender MAY modify or add to the request details but the payment request SHOULD serve as an Interledger packet without modification.
+
+### Operation
+
+The receiver creates an Interledger Packet from the details of the payment they want to request from the sender. The receiver uses a secret to create an HMAC of the packet. This HMAC is used as the condition fulfillment. The condition is the hash digest of the HMAC output.
+
+## Overview
+
+### Relation to Other Protocols
+
+     ------    
+    | SPSP |
+     ------ 
+       |
+     ------     -------
+    | ITP  |   |   ?   |
+     ------     -------
+       |           |
+     ------------------  
+    |       ILP        |
+     ------------------
+
+The [Simple Payment Setup Protocol (SPSP)](../0009-simple-payment-setup-protocol) uses the Interactive Transport Protocol (ITP). ITP is a host-to-host protocol intended for use on top of the Interledger Protocol.
+
+### Model of Operation
+
+We suppose that a sender wants to pay a receiver for some good or service and that the two are using a communication channel specified by a higher level protocol.
+
+We suppose that the receiver has a 32-byte secret value.
+
```

Comment:
Actually I think we're going to wait to add that feature later. I added an issue for it: https://github.com/interledger/rfcs/issues/53